### DISCUSSION TITLE
How do I save data in the free version

### DISCUSSION THREAD

**Discussion by goodpeti** [geoman-io/leaflet-geoman#858] [Questions & Help] [answered]
How do I save data to localStorage and/or online DataBase in the free version

**Comment by Falke-Design** [comment, marked as answer]
You can read out all layers on the map and save them as geojson to the localstorage or db.

```
var data = localStorage.getItem('data');
if(data){
	var geoLayer = L.geoJSON(JSON.parse(data)).addTo(map);
}


function saveToDb(){
	var allLayers = L.featureGroup();
        map.eachLayer(function(layer){
  	    if(layer instanceof L.Path || layer instanceof L.Marker){
    	      allLayers.addLayer(layer);
            }
         });
	var geojson = allLayers.toGeoJSON();
        localStorage.setItem('data', JSON.stringify(geojson));
}
```

But you have to add properties to the geojson which color, options the layer has and what type of layer it is. A Circle will displayed as Marker by default. But this is a default problem of geojson.

Example: https://jsfiddle.net/falkedesign/1z3kxadL/

**Comment by goodpeti** [comment]
Is it possible to upload the data to a MySQL database?

**Comment by codeofsumit** [comment]
@goodpeti you can upload the data to any database you like 😉 . That application logic is not part of this library though
